web/html/src/components/ranking-table.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from "react";
+import { DEPRECATED_unsafeEquals } from "utils/legacy";
+
+type Instance = JQuery & {};
+type Sortable = <T>(arg0: T, options?: any) => T extends string ? string[] : Instance;
+
+declare global {
+  interface JQuery {
+    sortable: Sortable;
+  }
+}
+
+function channelIcon(channel) {
+  let iconClass, iconTitle;
+  if (channel.type === "state") {
+    iconClass = "fa spacewalk-icon-salt-add";
+    iconTitle = t("State Configuration Channel");
+  } else {
+    iconClass = "fa spacewalk-icon-software-channels";
+    iconTitle = t("Normal Configuration Channel");
+  }
+
+  return <i className={iconClass} title={iconTitle} />;
+}
+
+type RankingTableProps = {
+  items: any[];
+  onUpdate?: (newItems: any[]) => any;
+  emptyMsg: string;
+};
+
+type RankingTableState = {
+  items: any[];
+};
+
+class RankingTable extends React.Component<RankingTableProps, RankingTableState> {
+  defaultEmptyMsg = t("There are no entries to show.");
+  node: Element | null = null;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: this.props.items,
+    };
+  }
+
+  cloneItems() {
+    return this.state.items.map(i => Object.assign({}, i));
```

Comment:
This is from the old existing code, just a heads up for consideration: this is not fully safe and will lead to unexpected results when applied to nested objects etc. Might not be relevant here though, I don't know what the data is.

Reply:
That's a good point. Data is a String for how we use it, but it could be anything we don't know in advance.
I'm about to refactor it using `_lodash.cloneDeep`, do you see any drawback?

```typescript
import _cloneDeep from "lodash/cloneDeep";
....
return this.state.items.map(i => _cloneDeep(i));
```

Reply:
Looks good to me. 👍

Reply:
Done